python-ecosys/urequests/urequests.py

```diff
@@ -33,7 +33,15 @@ def json(self):
         return ujson.loads(self.content)
 
 
-def request(method, url, data=None, json=None, headers={}, stream=None):
+def request(
+    method, url, data=None, json=None, headers={}, stream=None, parse_headers=True, auth=None
+):
+    redirect = None  # redirection url, None means no redirection
+    chunked = data and is_chunked_data(data)
+
+    if auth is not None:
+        headers.update(encode_basic_auth(auth[0], auth[1]))
```

Comment:
This is neat but does make me think that we should have at least some documentation (since it's not that obvious). I'll pull together a README, or at least suggestions what should be in it.